pycode/memilio-epidata/memilio/epidata/assessNPIEffects.py

```diff
@@ -287,24 +283,26 @@ def analyze_npi_data(
     # df_npis = mdfs.extract_subframe_based_on_dates(df_npis, date(2021,1,1), date(2021,6,1))
     npis = pd.read_json(os.path.join(directory, 'npis.json'))
     # get code levels (main/subcodes) and position of main codes
-    # code_level = [i.count('_') for i in npi_codes]
-    # main_code_pos = [i for i in range(len(code_level)) if code_level[i] == 1]
 
     # check if any other integer than 0: not implemented or 1: implemented is
     # used (maybe to specify the kind of implementation)
 
-    all_subcodes = [x for x in npis.NPI_code if len(x) <=8]
+    if not npi_codes_considered:
+        npi_codes_considered = [
+            x for x in npis[dd.EngEng['npiCode']] if len(x.split('_')) == 2]
     df_merged = df_npis.iloc[:, :2]
-    for subcode in all_subcodes:
+    for subcode in npi_codes_considered:
         # extract columns which have the subcode as part of the column
         # name and sum over all these subcodes
         df_merged[subcode] = df_npis.filter(
             regex=subcode).sum(axis=1)
-    #npi_codes_considered = npis.NPI_code.values.tolist()
-    npi_codes_considered = all_subcodes
-    df_npis = df_merged.copy()
 
-    if len(np.where(df_npis[npi_codes_considered] > 1)[0]) > 0:
+    # make sure that only considered subcodes are in npis dataframe
+    npis = npis[npis[dd.EngEng['npiCode']].isin(npi_codes_considered)]
+
+    df_npis = df_merged[:]
```

Comment:
is this a copy?

Reply:
It is a slice of the whole dataframe. With CoW this behaves as a copy but is faster